packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModuleManager.mm

```diff
@@ -808,9 +808,26 @@ - (Class)_getModuleClassFromName:(const char *)moduleName
     moduleClass = [_delegate getModuleClassFromName:moduleName];
   }
 
-  if (!moduleClass) {
-    moduleClass = getFallbackClassFromName(moduleName);
+  // Early return to avoid unnecessary computation
+  if (moduleClass != nil) {
+    return moduleClass;
+  }
+
+  moduleClass = getFallbackClassFromName(moduleName);
+  if (moduleClass != nil) {
+    return moduleClass;
   }
+
+  // fallback on modules registered throught RCT_EXPORT_MODULE with custom names
+  NSString *objcModuleName = [NSString stringWithUTF8String:moduleName];
+  auto modules = RCTGetModuleClasses();
```

Comment:
wondering when we should use auto? what's your criteria?

Reply:
lazyness! xD 
I didn't remember top of my head the type of `RCTGetModuleClasses` and then I forgot to use something better.

Reply:
BTW, I believe that `auto` can only be used in `.mm` files, as it is a C++ keyword.